plugins/optimization-detective/storage/data.php

```diff
@@ -140,6 +140,26 @@ function od_get_url_metrics_slug( array $query_vars ): string {
 	return md5( (string) wp_json_encode( $query_vars ) );
 }
 
+/**
+ * Computes the current ETag for URL Metrics.
+ *
+ * The ETag is a hash based on the IDs of the registered tag visitors
+ * in the current environment. It is used for marking the URL Metrics as stale
+ * when its value changes.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param OD_Tag_Visitor_Registry $tag_visitor_registry Tag visitor registry.
+ * @return string Current ETag.
+ */
+function od_compute_current_etag( OD_Tag_Visitor_Registry $tag_visitor_registry ): string {
+	$data = array(
+		'tag_visitors' => array_keys( iterator_to_array( $tag_visitor_registry ) ),
+	);
+	return md5( serialize( $data ) );  // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
```

Comment:
Perhaps we should JSON encode this?

Reply:
Yeah, since `od_get_url_metrics_slug()` is using it:

```suggestion
	return md5( (string) wp_json_encode( $data ) );
```